ArticleController routes that does a simple article search and show page (no paging, facets, etc.)

drh-stanford opened the pull request:
This PR fixes #1380. It implements a basic article search and show page without Blacklight features like paging, facets, etc. To use, you need to add the EDS_* settings to your `config/settings/development.local.yml` (see Slack).

Notes:

- I marked the `search` test for the `Eds::Repository` as pending since the mocking for this is a little detailed. Similarly, I marked the feature spec for article search as pending -- we need a way to mock EDS API results basically. The views are also not spec'd as they are vanilla blacklight partials right now.
- Not all of the EDS metadata is shown -- they are working on providing us with a complete list of the metadata available. Notably, the links to full text, etc. are not in this PR.
- Not all the functionality in `Eds::Repository.send_and_receive` is being used right now -- specifically, the next/prev and multiple id fetching support.

### Search results
<img width="747" alt="screen shot 2017-06-23 at 3 28 43 pm" src="https://user-images.githubusercontent.com/1861171/27502418-c021f884-5828-11e7-89e7-687a422f31f8.png">



### Show page

<img width="761" alt="screen shot 2017-06-23 at 3 29 02 pm" src="https://user-images.githubusercontent.com/1861171/27502422-c6ce6866-5828-11e7-9254-d7f248e690e1.png">

jkeck commented on app/controllers/article_controller.rb:
```diff
@@ -2,5 +2,80 @@
+    session['guest'] = false # TODO: hardcoded to bypass auth
+    session['eds_session_token'] =
+      EBSCO::EDS::Session.new(
+        guest: false, # TODO: hardcoded to bypass auth
```
Wondering if we should make this true by default, even for our initial implementation.  Currently, anybody on campus could theoretically access this site if they knew about it, and could potentially access metadata if items they are technically not supposed to be able to.

drh-stanford replied:
Ah, good point. I'll make it guest-only.

jkeck commented on app/helpers/article_helper.rb:
```diff
@@ -2,4 +2,19 @@ module ArticleHelper
+    doi = options[:value].try(:first)
+    return if doi.blank?
+    url = 'https://doi.org/' + doi.to_s
+    link_to(doi, url, target: '_blank')
```
We typically try to avoid `_blank` targets for usability purposes.  @jvine any reason to keep this as a `target="_blank"`?

drh-stanford replied:
I can remove it -- the only reason it's there is because that's how the EDS example helper had it.

jkeck commented on config/initializers/eds.rb:
```diff
@@ -0,0 +1,3 @@
+require 'ebsco/eds'
+
+fail unless Settings.EDS_USER && Settings.EDS_PASS && Settings.EDS_PROFILE
```
Can we provide some sort of message about why we're failing here?

jkeck commented on spec/controllers/article_controller_spec.rb:
```diff
@@ -7,4 +7,27 @@
   end
+
+  it 'handles authentication'
+  it 'handles configuration'
```
Are these `TODO` tests?  (I also see a few others in other places, just not familiar w/ this pattern)

drh-stanford replied:
yes these are TODO/pending tests

jkeck commented:
I _think_ you can fix that failing test by rebasing on-top of master.  That test should have been fixed in #1390.

drh-stanford commented:
Ok, I think I've addressed all the feedback. Note that the coverage went down due to pending specs, in particular the ones where we need to mock the EDS API.
